### PR TITLE
Refactor variant and genotype annotations

### LINKED ISSUE #89: Streamline genotype/variant annotations
* Rename `DatabaseVariantAnnotation` to `VariantAnnotation`
* Rename `VariantCallingAnnotations` to `GenotypeAnnotation`
* Roll `TranscriptEffect` under `VariantAnnotation`.

### PR DESCRIPTION
**Pull request by heuermh**
Fixes #89 

Also includes style fixes, removes used Base enum, and removes unused sample fields.

### CODE REVIEW

File: src/main/resources/avro/bdg.avdl [hunk 1]
```diff
@@ -924,54 +900,116 @@ record TranscriptEffect {
...
+   */
+  union { null, int } reverseReadDepth = null;
 
-  //conservation
-  union { null, string } gerpNr  = null;
-  union { null, string } gerpRs  = null;
-  union { null, float } phylop  = null;
-  union { null, string } ancestralAllele  = null;
+  /**
+   Allele frequency, VCF INFO header line reserved key AF.
```
Comment (fnothaft) [inline src/main/resources/avro/bdg.avdl]: A random note: we should name this either major or minor allele frequency.

Comment (fnothaft) [inline src/main/resources/avro/bdg.avdl]: Also, I looked this up and VCF AF is minor allele frequency.

Comment (heuermh) [inline src/main/resources/avro/bdg.avdl]: Thanks, will update the doc

**Comment by heuermh** [comment]
Thanks!

**Comment by fnothaft** [comment]
Merged! Thanks @heuermh!

**Pull request merged**